### ISSUE TITLE
intermittent issue: "Unsupported signal: <nil>. Discarding.." pending to the return from "docker run" on latest windows docker CE master-dockerproject-2021-02-03

### ISSUE THREAD

**Issue by bingbing8** [moby/moby#41985]
**Description**
intermittent issue: "Unsupported signal: <nil>. Discarding.." pending to the return from "docker run" on latest windows docker from master.dockerproject.org

intermittent issue: "Unsupported signal: <nil>. Discarding.." pending to the return from "docker run" on latest windows docker from master.dockerproject.org

**Steps to reproduce the issue:**
  1. C:\docker pull mcr.microsoft.com/windows/nanoserver:20H2-KB4598242
20H2-KB4598242: Pulling from windows/nanoserver
2ddfbf39b1d8: Already exists
Digest: sha256:b0a754e60c8fdf311b29db7d871452a431c4b86ae4730450b6fe320da533e31e
Status: Downloaded newer image for mcr.microsoft.com/windows/nanoserver:20H2-KB4598242
mcr.microsoft.com/windows/nanoserver:20H2-KB4598242
  2. C:\>docker run --isolation hyperv --rm mcr.microsoft.com/windows/nanoserver:20H2-KB4598242 cmd /k echo hello
hello

C:\>Unsupported signal: <nil>. Discarding.

C:\>docker run --rm mcr.microsoft.com/windows/nanoserver:20H2-KB4598242 cmd /k echo hello
hello

C:\>Unsupported signal: <nil>. Discarding.

**Describe the results you received:**
"Unsupported signal: <nil>. Discarding." is pending to the returned expected message

**Describe the results you expected:**

no pending error
**Additional information you deem important (e.g. issue happens only occasionally):**
Container host Windows Version: it repros on any version of windows host both server and client sku.  server2019, Windows Server 1909, Windows Server 1903, Windows Server 20H2, Windows Server 2004
  OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.19042 N/A Build 19042

I can repro it inside a VM as container host, but did not repro on physical machine.
**Output of `docker version`:**

```
docker version
Client: Docker Engine - Community
 Version:           master-dockerproject-2021-02-03
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        70a00157f1
 Built:             Wed Feb  3 23:58:43 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          master-dockerproject-2021-02-03
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.13.15
  Git commit:       8d31795
  Built:            Thu Feb  4 00:08:25 2021
  OS/Arch:          windows/amd64
  Experimental:     true
```

**Output of `docker info`:**

```
docker info
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 5
 Server Version: master-dockerproject-2021-02-03
 Storage Driver: windowsfilter (windows) lcow (linux)
  Windows:
  LCOW:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 19042 (19041.1.amd64fre.vb_release.191206-1406)
 Operating System: Windows 10 Enterprise Version 2009 (OS Build 19042.746)
 OSType: windows
 Architecture: x86_64
 CPUs: 8
 Total Memory: 16GiB
 Name: T20h22101BC
 ID: UC7Q:3NRT:BELW:XWDW:43UT:PIMS:JWTW:AA55:JVBY:KWCQ:OAWQ:QXUG
 Docker Root Dir: C:\docker
 Debug Mode: true
  File Descriptors: -1
  Goroutines: 25
  System Time: 2021-02-04T18:51:25.9398485-08:00
  EventsListeners: 0
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

WARNING: API is accessible on http://0.0.0.0:2375 without encryption.
         Access to the remote API is equivalent to root access on the host. Refer
         to the 'Docker daemon attack surface' section in the documentation for
         more information: https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows VM

**Comment by thaJeztah** [comment]
looks related to https://github.com/docker/cli/pull/2987 (which fixes this in 20.10.5)

and being worked on (for master) in https://github.com/docker/cli/pull/2986

**Comment by bingbing8** [comment]
Tried on this version, still got repro:
Client:
Version:           master-dockerproject-2021-05-04
API version:       1.41
Go version:        go1.16.3
Git commit:        cde06fbf1d
Built:             Wed May  5 00:07:18 2021
OS/Arch:           windows/amd64
Context:           default
Experimental:      true
Server: Docker Engine - Community
Engine:
  Version:          master-dockerproject-2021-05-04
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.13.15
  Git commit:       892d3d5
  Built:            Wed May  5 00:03:14 2021
  OS/Arch:          windows/amd64
  Experimental:     true

**Comment by AmirHosseinSaadatmand** [comment]
I had this issue. this error occurred because of CMD runtime error in my Dockerfile.

**Comment by thaJeztah** [comment]
This should be fixed on cli "master" through https://github.com/docker/cli/pull/2986, and is being back ported to the 20.10 release branch (used for docker desktop)  through https://github.com/docker/cli/pull/3107, which will be included in the 20.10.7 patch release
